File: client-app/src/features/AppAttachment/AppAttachment.ts
export interface IAppAttachment {
	Id: number
	AppDataId: number
	FileName: string
	Path: string
	Prop1: string
	Prop2: string
	Prop3: string
	Prop4: string
	Prop5: string
}

export class AppAttachment implements IAppAttachment {
	Id: number = 0;
	AppDataId: number = 0;
	FileName: string = '';
	Path: string = '';
	Prop1: string = '';
	Prop2: string = '';
	Prop3: string = '';
	Prop4: string = '';
	Prop5: string = '';
  
  constructor(init?: IAppAttachment) {
    Object.assign(this, init);
  }
}


export interface IApiAttachment {
	File: Blob;		
	FileName?: string
	Prop1?: string
	Prop2?: string
	Prop3?: string
	Prop4?: string
	Prop5?: string
}

export class ApiAttachment implements IApiAttachment {
	File: Blob = new Blob();	
	FileName: string = '';	
	Prop1: string = '';
	Prop2: string = '';
	Prop3: string = '';
	Prop4: string = '';
	Prop5: string = '';
  
  constructor(init?: IApiAttachment) {
    Object.assign(this, init);
  }
}

File: Domain/AppAttachment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public class AppAttachment
    {
		[Key]
      	[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public int Id { get; set; }
		public int AppDataId { get; set; }
		public int AppDataColumn { get; set; }
		public string FileName { get; set; }
		public string Path { get; set; }
		public string Prop1 { get; set; }
		public string Prop2 { get; set; }
		public string Prop3 { get; set; }
		public string Prop4 { get; set; }
		public string Prop5 { get; set; }
    }
}

















